server/prisma/migrations/20240131062714_update_order_process/migration.sql

```diff
@@ -0,0 +1,212 @@
+ALTER TABLE "orders" DROP CONSTRAINT "orders_orderdate_fkey";
+
+-- DropForeignKey
+ALTER TABLE "seasontickets" DROP CONSTRAINT "seasontickets_eventid_fk_fkey";
+
+-- DropForeignKey
+ALTER TABLE "seasontickets" DROP CONSTRAINT "seasontickets_eventticketid_fk_fkey";
+
+-- DropForeignKey
+ALTER TABLE "seasontickets" DROP CONSTRAINT "seasontickets_orderitemid_fk_fkey";
+
+-- DropForeignKey
+ALTER TABLE "seasontickets" DROP CONSTRAINT "seasontickets_seasontickettypeid_fk_fkey";
+
+-- DropForeignKey
+ALTER TABLE "singletickets" DROP CONSTRAINT "singletickets_orderitemid_fk_fkey";
+
+-- AlterTable
+ALTER TABLE "donations" DROP COLUMN "contactid_fk",
+DROP COLUMN "donationdate",
+DROP COLUMN "donorname",
+DROP COLUMN "isanonymous",
+DROP COLUMN "payment_intent",
+DROP COLUMN "refund_intent",
+ADD COLUMN     "anonymous" BOOLEAN NOT NULL DEFAULT false,
+ADD COLUMN     "orderid_fk" INTEGER NOT NULL,
+ALTER COLUMN "amount" SET NOT NULL,
+ALTER COLUMN "frequency" SET NOT NULL;
+
+-- AlterTable
+ALTER TABLE "eventinstances" DROP COLUMN "defaulttickettype";
+
+-- AlterTable
+ALTER TABLE "events" ALTER COLUMN "eventdescription" SET NOT NULL,
+ALTER COLUMN "eventdescription" SET DEFAULT '',
+ALTER COLUMN "active" SET NOT NULL,
+ALTER COLUMN "active" SET DEFAULT true,
+ALTER COLUMN "seasonticketeligible" SET NOT NULL,
+ALTER COLUMN "seasonticketeligible" SET DEFAULT true;
+
+-- AlterTable
+ALTER TABLE "orders" DROP COLUMN "orderdate",
+DROP COLUMN "ordertime",
+DROP COLUMN "ordertotal",
+DROP COLUMN "refund_intent",
+ADD COLUMN     "orderdateandtime" TIMESTAMP(3) NOT NULL DEFAULT CURRENT_TIMESTAMP;
+
+-- AlterTable
+ALTER TABLE "seasons" ADD COLUMN     "deletedat" TIMESTAMP;
+
+-- AlterTable
+ALTER TABLE "ticketrestrictions" ADD COLUMN     "deletedat" TIMESTAMP(3);
+
+-- DropTable
+DROP TABLE "eventtickets";
+
+-- DropTable
+DROP TABLE "orderitems";
+
+-- DropTable
+DROP TABLE "seasontickets";
+
+-- DropTable
+DROP TABLE "seasontickettype";
+
+-- DropTable
+DROP TABLE "singletickets";
+
+-- CreateTable
+CREATE TABLE "order_ticketitems" (
+    "id" SERIAL NOT NULL,
+    "orderid_fk" INTEGER NOT NULL,
+    "price" MONEY NOT NULL,
+
+    CONSTRAINT "order_ticketitems_pkey" PRIMARY KEY ("id")
+);
+
+-- CreateTable
+CREATE TABLE "ticketitems" (
+    "id" SERIAL NOT NULL,
+    "order_ticketitemid_fk" INTEGER NOT NULL,
+    "ticketrestrictionid_fk" INTEGER NOT NULL,
+    "donated" BOOLEAN NOT NULL DEFAULT false,
+    "redeemed" TIMESTAMP(3),
+
+    CONSTRAINT "ticketitems_pkey" PRIMARY KEY ("id")
+);
+
+-- CreateTable
+CREATE TABLE "refunds" (
+    "id" SERIAL NOT NULL,
+    "orderid_fk" INTEGER NOT NULL,
+    "refund_intent" VARCHAR(255) NOT NULL,
+
+    CONSTRAINT "refunds_pkey" PRIMARY KEY ("id")
+);
+
+-- CreateTable
+CREATE TABLE "refunditems" (
+    "id" SERIAL NOT NULL,
+    "refundid_fk" INTEGER NOT NULL,
+    "order_ticketitemid_fk" INTEGER,
+    "donationid_fk" INTEGER,
+    "amount" MONEY NOT NULL,
+
+    CONSTRAINT "refunditems_pkey" PRIMARY KEY ("id")
+);
```

Comment:
What's the difference between these? Why do they need to be distinct tables? Presumably there is the case where we would want to be able to partially refund an order, would that result in multiple `refund_intents` from stripe? Then we would want to keep track of each one and the associated amount

Reply:
Currently the schema doesn't support multiple refunds per order, this update would allow for zero to many refunds per order with each refund having one to many refund items. A refund Item would contain the id of the item being refunded as well as the amount of  the original price that was refunded for that individual item (I was thinking this would allow for partial refunds on an individual item level if that ever was a need down the line).

Reply:
I see, I think we could still get away with flattening these, but since we already have a pattern of smaller tables mapped with lots of relations I think this is fine.